### PR TITLE
feat: Add products listing in homepage

### PR DESCRIPTION
**Pull request by Gowreesh-A-M**

### CODE REVIEW

File: macos/Flutter/GeneratedPluginRegistrant.swift
Comment (AswinRaaj23) [inline macos/Flutter/GeneratedPluginRegistrant.swift:1]: Not relevant to this PR

File: lib/widgets/product_grid.dart
Comment (AswinRaaj23) [inline lib/widgets/product_grid.dart:1]: This can be a method in home_screen itself

File: lib/widgets/product_card.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  Widget build(BuildContext context) {
+    return Card(
+      elevation: 4,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(12),
+      ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          _buildImage(),
```
Comment (AswinRaaj23) [inline lib/widgets/product_card.dart]: be consistent in naming

_buildProductImage()
_buildProductPrice()

**Review by AswinRaaj23 — APPROVED** [review]

File: lib/widgets/product_grid.dart
Comment (Gowreesh-A-M) [reply lib/widgets/product_grid.dart:1]: I thought of reusing it bro, as there are tasks like the Wishlist page and category filters on my to-do list. I also felt that the ProductGrid widget has a different responsibility compared to the HomeScreen widget, so making it a separate widget will be beneficial in either way

File: macos/Flutter/GeneratedPluginRegistrant.swift
Comment (Gowreesh-A-M) [reply macos/Flutter/GeneratedPluginRegistrant.swift:1]: when I added the CachedNetworkImage dependency it automatically generated bro. I thought it was a part of the process. Should I have to push that file alone in the main before the PR?

File: lib/widgets/product_card.dart (hunk 1 above)
Comment (Gowreesh-A-M) [reply lib/widgets/product_card.dart]: ok bro

**Pull request merged**